crate/server_database/src/stores/redis/permissions.rs

```diff
@@ -66,266 +105,180 @@ impl Triple {
     }
 }
 
-impl TryFrom<&Location> for Triple {
+impl Serializable for PermTriple {
     type Error = DbError;
 
-    fn try_from(value: &Location) -> Result<Self, Self::Error> {
-        let value = String::from_utf8((value).to_vec())?;
-        let mut parts = value.split("::");
-        let uid = parts.next().ok_or_else(|| {
-            DbError::ConversionError(format!("invalid permissions triple: {parts:?}"))
-        })?;
-        let user_id = parts.next().ok_or_else(|| {
-            DbError::ConversionError(format!("invalid permissions triple: {parts:?}"))
-        })?;
-        let permission = parts.next().ok_or_else(|| {
-            DbError::ConversionError(format!("invalid permissions triple: {parts:?}"))
+    fn length(&self) -> usize {
+        // obj_uid + user_id + permission
+        self.obj_uid.0.length() + self.user_id.0.length() + 1 // 1 byte is KmipOperation's enum size
+    }
+
+    fn write(
+        &self,
+        ser: &mut cosmian_kms_crypto::reexport::cosmian_crypto_core::bytes_ser_de::Serializer, // full dependency path spec if needed to avoid collisions
+    ) -> Result<usize, Self::Error> {
+        let mut written = 0;
+        written += ser.write(&self.obj_uid.0)?;
+        written += ser.write(&self.user_id.0)?;
+        written += ser.write_vec(&[self.permission as u8])?;
+        Ok(written)
+    }
```

Comment:
Did you check your serialization with `test_serialization` ? I am pretty sure your length is not in sync with your `write` implementation. I wouldn't use `write_vec` but rather `ser.write(self.permission as u8)` if `Serializable` is implemented for `u8` in `cosmian_crypto_core` and implement it there if it is not.